seek for seq (and rseq)

FiV0 opened the pull request:
This is a first cut of the `seek` op for seqs of PersistentSortedSets. I haven't done anything for CLJS yet, wanted to get some feedback first before spending more time on it. 

If you want me to the PR against your `storage` branch, let me know.

FiV0 committed:
seek for seq (and rseq)

FiV0 commented on src-java/me/tonsky/persistent_sorted_set/Seq.java:
```diff
@@ -156,4 +156,74 @@ public Seq rseq() {
+
+    } else {
+
+      while (cmp.compare(to, minKey(node)) < 0){
```
This is a bit `awkward` here. There is currently no way to get the minimum key of any of it's descendants of a `Node` in constant time. This should probably be somehow made possible o/w `seek` for rseq will have a lot of unnecessary overhead.

tonsky commented:
Looks good to me. But I suggest you rebase it on `storage` branch or wait for it to be merged (this week I hope).

Re: rseek, the only problem is that we go one level too high compared to what we would’ve done if we had fast min-key, no? If we replace `cmp.compare(to, minKey(node)) < 0` with `cmp.compare(to, keys[0]) < 0`. Or wouldn’t that work?

FiV0 committed:
Don't shortcircuit for rseek

FiV0 commented:
Re:Re:rseek Right. In that case we can't shortcircuit, but I guess that is not too bad.

FiV0 committed:
Node -> Branch

FiV0 committed:
ISeek for cljs

FiV0 commented:
@tonsky I rebased and added the cljs part. As there are no parent pointers in the cljs implementation I essentially start from the root every time I seek. This is of course not as efficient as walking the tree, but I hope good enough for cljs.

FiV0 committed:
seek section for README

tonsky commented:
Merged and released 0.2.1. Thank you!
